src/mjolnir/ferry_connections.cc

```diff
@@ -119,7 +119,7 @@ uint32_t ShortestPath(const uint32_t start_node_idx,
 
       // Skip uses other than road / other (service?)
       const OSMWay w = *ways[edge.wayindex_];
-      if (w.use() != baldr::Use::kOther &&
+      if (w.use() != baldr::Use::kOther && w.use() != baldr::Use::kServiceRoad &&
```

Comment:
Is Use::kOther needed now that this explicitly looks for service road use?

Reply:
@SvetlanaBayarovich we should check on this.  looking at https://github.com/valhalla/valhalla/blob/master/lua/graph.lua#L1332 if not one of the services above and it is not empty it will be set to other

Reply:
But this code above:

```
elseif use == nil and kv["highway"] == "service" then
        use = 11
```

is what makes highway=service (without the explicit service=... in the use method) have use = 11 (kServiceRoad).

Reply:
If I understand correctly, the whole picture is like 
```
  if kv["highway"] then
     if
     ...
     elseif use == nil and kv["highway"] == "service" then
        use = 11
     ...    
     end
  end

  if use == nil and kv["service"] then
    use = 40 --other
  ...
```
So we only get assigned use = 40 if there is a `service` tag but no `highway` in the item (way?) being parsed. I don't have a feeling if it's possible for the OSM data. On one hand, looks like any combination of tags is theoretically possible, but maybe there are some constraints that I'm not aware of.

@gknisely  @dnesbitt61 Is there a way to understand that? My only idea at the moment is to run over all the edges created and check if use = 40 was ever assigned

Reply:
But still such check guarantees nothing for future

Reply:
I do not know if this tagging combination happens. Perhaps taginfo could reveal if this combination is possible? I know that service=driveway says that highway=service is required but I don't know if there is anything that prevents a service=XXXX without highway=service. It probably can't hurt to keep the Use::kOther for cases that don't fall into any of the existing service uses (generic, alley, driveway, parking_aisle).

Do you think that Use::kServiceRoad is missing on some edges leading to ferries?

Reply:
I'm not sure I got your question...
But I have a related question. There is a case when one can get to a ferry only if one uses a drive-through edge. And it obviously gets Use::kDriveThru. Currently such edges are also excluded from connection search. Is it legal to use such label here at all?
![image](https://user-images.githubusercontent.com/50918075/111360400-c0f1ed80-869d-11eb-982b-5c5babcebf11.png)

Reply:
What I was trying to ask is whether you were seeing some roads that were not properly marked as Use::kServiceRoad leading to a ferry. Sounds like you found a case with highway:service and service:drive-through that should be reclassified as it leads to a ferry.

https://wiki.openstreetmap.org/wiki/Tag:service%3Ddrive-through - That seems to indicate that perhaps this other service road type needs to be considered when reclassifying edges leading to ferries. I imagine this is a case where "where customers can visit a business without leaving their cars." (the business is the ferry itself).

Reply:
Hmm, maybe I'd rather have this change as a separate update in order not to slow down this pr which actually only brings back what was already there, as I think it'll take some time to test

Reply:
makes sense to me